.github/workflows/on-push.yaml

```diff
@@ -1,30 +1,45 @@
 name: on push
 
-on: [push]
-    
+on:
+  push:
+    branches:
+      - main
+      - refactor/tests
+
+env:
+  PYTHON_VERSION: 3.9
+
 jobs:
   tests:
     name: Tests
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      
+
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: '3.x'
-      
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install dependencies
+        working-directory: ./src
+        run: pip install -r requirements.txt -r requirements_dev.txt
+
+      - name: Lint
+        working-directory: ./src
+        run: |
+          isort --check-only .
+          black --check .
+
       - name: Tests
+        working-directory: ./src
         run: |
           for year in $(seq -w 2015 2023); do
             for day in $(seq -w 1 25); do
               if [ -d "$year/day$day" ]; then
                 echo "----------------------------------------------------------------------"
                 echo -e "\033[0;32mRunning tests for $year day $day ..."
-                cd "$year/day$day" || exit
-                python -m unittest tests.py
-                cd ../.. || exit
+                pytest -v --no-header $year/day$day
```

Comment:
🏆

Reply:
In my repo, i have these as a bash script so that i can also run it locally if i want. And in gha, i just run the bash script. Also makes it easy to test changes to the bash scripts 😅

Reply:
makes sense! the same way to run locally and on pipelines, since I'm using a `makefile`, I've moved this to the `makefile`. 

Thanks a lot for the review, @rpidanny!